src/game/game.py

```diff
@@ -5,7 +5,59 @@
 class Game(polymodel.PolyModel):
     """Represents a single instance of a Game"""
     date = ndb.DateTimeProperty(auto_now_add=True, required=True)
-    player_keys = ndb.KeyProperty(kind="User", repeated=True, required=True)
+    player_keys = ndb.KeyProperty(kind="User", repeated=True)
     player_record_keys = ndb.KeyProperty(
-        kind="GamePlayerRecord", repeated=True, required=True)
+        kind="GamePlayerRecord", repeated=True)
     winner_keys = ndb.KeyProperty(kind="User", repeated=True)
+
+    @property
+    def player_records(self):
+        if hasattr(self, "_player_records"):
+            return self._player_records
+        self._player_records = ndb.get_multi(self.player_record_keys)
+        return self._player_records
+
+    @classmethod
+    def add(cls, player_records):
+        """
+        Use a game type and a list of player records to add a Game to db
+        :param player_records a list of GamePlayerRecords
+        :returns: a game object
+        """
+        game = cls(player_record_keys=[r.key for r in player_records])
+        game.calculate_experience()
+        game.calculate_rating_change()
+        game.players = [s.player for s in player_records]
+        game.put()
+        return game  #  TODO: Calculate winner
+
+    def calculate_rating_change(self):
+        """
+        Calculate points using self.player_records
+        Should be overwritten by subclass
+        """
+        pass
+
+    def calculate_experience(self):
+        """
+        Use self.player_records to assign experience to the relevant players
+        """
+        pass
+
+    @staticmethod
+    def _update_ranked_data(sorted_player_records, rated_rating_groups):
+        to_put = sorted_player_records
+        record_index = 0
+        for team in rated_rating_groups:
+            for rating in team.itervalues():
+                record = sorted_player_records[record_index]
+                player = record.player
+
+                points_earned = rating.mu - player.rank_data.points
```

Comment:
I disagree - the top of the ws is getting the record and player, then the next 2 lines are updating those variables.